### PR TITLE
Fix white space queries breaking the editor

### PR DESCRIPTION
**Pull request by arthur801031**
Associated Issue: #1930

### Summary of Changes
* Add a regex to check if the query only contains whitespace. If this is true, we do not search this query. Otherwise, we perform a search.

### CODE REVIEW

**Comment by arthur801031** [comment]
Hi @jasonLaster 

I have a hack that fixes this issue. The editor would break when the query only contains whitespace(s), so I wrote a function to check if the query only contains whitespace(s). If this is true, we do not perform a search; otherwise, a search is performed.

In this example, " DOM" and "DOM ", query that contains leading whitespace or trailing whitespace and other characters still works.
3dlxtti1x6

File: src/components/Editor/SearchBar.js [hunk 1]
```diff
@@ -152,7 +152,18 @@ const SearchBar = React.createClass({
...
     // eslint-disable-next-line react/no-did-update-set-state
     this.setState({ count, index: 0 });
-    this.search(query);
+    if (!this.leadingWhiteSpaceOnly(query)) {
+      this.search(query);
+    }
+  },
+
+  leadingWhiteSpaceOnly(query: string) {
+    for (let i = 0; i < query.length; i++) {
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: Hmm. Maybe use a regex?

Comment (arthur801031) [inline src/components/Editor/SearchBar.js]: Sure!

**Comment by jasonLaster** [comment]
@arthur801031 i moved the check to `source-search` so that the logic is more centralized.

@wldcordeiro @bomsy what do you think?

**Comment by bomsy** [comment]
Looks Good!

**Comment by wldcordeiro** [comment]
Maybe adding a test as well for this case would help in a follow-up PR?

**Comment by jasonLaster** [comment]
Yea. Integration tests coming

**Pull request merged**